### ISSUE TITLE
Image URL as a web component attribute 

### ISSUE THREAD

**Issue by gdbaldw** [modernweb-dev/rocket#162]
My web component takes an image URL as an attribute. In dev mode, all works perfectly. But, Rocket build renames the image URL, and does not update the web component's attribute. Is there a filter to update the image URL? Or a configuration to pass through images without changing the URL?

**Comment by daKmoR** [comment]
so you have something like 

```html
<my-el img="./path/to/file.jpg"></my-el>
```

does the file get copied over?
what is the output in build and what is the actual file URL?

**Comment by gdbaldw** [comment]
Yes. Though in my case it data from markdown front matter, and the data is absolute from root.

```
  ---
  image_url: /foo/bar.jpg
  ---
```

and my njk template is like this

```javascript
  <my-el image_url="{{ image_url | url }}"></my-el>
  <img src="{{ image_url | url }}">
```

The img element is correct with the file copied and renamed, and my-el is...
```javascript
  <my-el image_url="/foo/bar.jpg"></my-el>
  <img src="xxxxx.jpg">
```

I am recalling from memory, and will recheck the details later today.

**Comment by gdbaldw** [comment]
So, I tried what you may be hinting, using `src` as attribute name.

```javascript
  <my-el src="{{ image_url | url }}"></my-el>
  <img src="{{ image_url | url }}">
```
Same result, but now also have reference errors.

```
Error: ❌ Found 2 missing reference targets (used by 2 links) while checking 14 files
```

**Comment by daKmoR** [comment]
ah so you use it in two places

```html
  <!-- a custom tag will not be checked or modified -->
  <my-el image_url="{{ image_url | url }}"></my-el>
  
  <!-- this will find the image and copy/hash and adjust this src -->
  <img src="{{ image_url | url }}"> 
```

so if you only use `<my-el image_url="{{ image_url | url }}"></my-el>` then it should not be copied/hashed or adjusted...

now if you were to put the image into `docs/_assets/_static/test.jpg` and then insert it like so
```html
<my-el src="{{ image_url | asset | url }}"></my-el>
```

then I think it should work as all `_static` images are copied as is and the URL would then not be touched. This however means if you use it as a normal image somewhere as well that it will be duplicated once renamed and hashed once statically copied


Ideally, you would be able to adjust which tag and which attribute gets defined as an asset... but that API is currently not there 😅.
the tools used is https://modern-web.dev/docs/building/rollup-plugin-html/

**Comment by gdbaldw** [comment]
Thank you for the clarification. Maybe, for rollup-plugin-html...

params.extractAssets: boolean | { tag: string,  attributes: string[] }[]

Case statements in the plugin src might be preceded by a params.extractAssets.forEach

It  a minor issue for me now, so I'm monkey patching attributes in my _site files.

**Comment by gdbaldw** [comment]
Attempting to update https://github.com/modernweb-dev/web/tree/master/packages/rollup-plugin-html, but a clean install and test has errors. Not sure how to get a clean baseline to work from.

```
yarn test
yarn run v1.22.10
$ mocha test/**/*.test.ts --require ts-node/register

TSError: ⨯ Unable to compile TypeScript:
src/input/extract/extractModules.ts:1:10 - error TS2305: Module '"../../../../../node_modules/@web/parse5-utils"' has no exported member 'findElements'.

1 import { findElements, getAttribute, getTagName, getTextContent, remove } from '@web/parse5-utils';
           ~~~~~~~~~~~~
src/input/extract/extractModules.ts:1:24 - error TS2305: Module '"../../../../../node_modules/@web/parse5-utils"' has no exported member 'getAttribute'.

1 import { findElements, getAttribute, getTagName, getTextContent, remove } from '@web/parse5-utils';
                         ~~~~~~~~~~~~
src/input/extract/extractModules.ts:1:38 - error TS2305: Module '"../../../../../node_modules/@web/parse5-utils"' has no exported member 'getTagName'.

1 import { findElements, getAttribute, getTagName, getTextContent, remove } from '@web/parse5-utils';
                                       ~~~~~~~~~~
src/input/extract/extractModules.ts:1:50 - error TS2305: Module '"../../../../../node_modules/@web/parse5-utils"' has no exported member 'getTextContent'.

1 import { findElements, getAttribute, getTagName, getTextContent, remove } from '@web/parse5-utils';
                                                   ~~~~~~~~~~~~~~
src/input/extract/extractModules.ts:1:66 - error TS2305: Module '"../../../../../node_modules/@web/parse5-utils"' has no exported member 'remove'.

1 import { findElements, getAttribute, getTagName, getTextContent, remove } from '@web/parse5-utils';
                                                                   ~~~~~~
src/input/extract/extractModules.ts:6:10 - error TS2305: Module '"../../../../../node_modules/@web/parse5-utils"' has no exported member 'getAttributes'.

6 import { getAttributes } from '@web/parse5-utils';
           ~~~~~~~~~~~~~
src/input/extract/extractModules.ts:33:5 - error TS7006: Parameter 'e' implicitly has an 'any' type.

```

**Comment by gdbaldw** [comment]
Pull request https://github.com/modernweb-dev/web/pull/1541 submitted. The request is intended to close this issue (and save me from monkey patching my _site files).

**Comment by daKmoR** [comment]
Thanks for the detailed report and follow-up.

This issue belongs to the old Rocket asset pipeline, where build-time HTML processing rewrote known asset attributes like `<img src>`, but custom element attributes such as `<my-el image_url="...">` were not treated as asset references.

Rocket has since been rebuilt from a clean slate as `@rocket/js`, with a different page/component architecture and build pipeline. The old behavior no longer maps cleanly to the current codebase, so I’m closing this as part of the repository reset.

The underlying use case is still valid though: custom elements often need asset URLs as attributes or properties. In the new Rocket, that should be handled explicitly through source code/imports, page data, static assets, or a new documented asset helper/API rather than by mutating arbitrary custom element attributes during HTML post-processing.

If this is still needed in the new version, please open a fresh issue with a current `@rocket/js` example. A good new issue would be something like:

“Support/Document asset URLs for custom element attributes”

Thanks again for surfacing this — it is exactly the kind of edge case the new docs should explain clearly.
